lib/index.js

```diff
@@ -20,131 +19,49 @@ internals.schema = {
         preRegister: Joi.func().allow(false)
     }),
     manifest: Joi.object({
-        server: Joi.object(),
-        connections: Joi.array().items(Joi.object()),
-        registrations: Joi.array().items(Joi.object({
-            plugin: [
-                Joi.string(),
-                Joi.object({ register: [Joi.string(), Joi.func()] }).unknown()
-            ],
-            options: Joi.object()
-        }))
+        server  : Joi.object(),
+        register: Joi.object({
+            plugins: Joi.array(),
+            options: Joi.any()
+        })
     })
 };
 
 
-exports.compose = function (manifest /*, [options], [callback] */) {
-
-    Hoek.assert(arguments.length <= 3, 'Invalid number of arguments');
-
-    const options = arguments.length === 1 || typeof arguments[1] === 'function' ? {} : arguments[1];
-    const callback = typeof arguments[arguments.length - 1] === 'function' ? arguments[arguments.length - 1] : null;
+exports.compose = async function (manifest, options = {}) {
 
     Joi.assert(options, internals.schema.options, 'Invalid options');
     Joi.assert(manifest, internals.schema.manifest, 'Invalid manifest');
 
-    // Return Promise if no callback provided
-
-    if (!callback) {
-        return new Promise((resolve, reject) => {
-
-            exports.compose(manifest, options, (err, server) => {
-
-                if (err) {
-                    return reject(err);
-                }
-
-                return resolve(server);
-            });
-        });
-    }
-
     // Create server
```

Comment:
might as well drop this comment - there are no other comments, and its not really pertinent to call out.